src/slot_machine.c

```diff
@@ -1978,196 +2153,221 @@ static bool8 AwardPayoutAction_FreeTask(struct Task *task)
+#define tTimer          data[2]
+#define tMoving         data[14]
+#define tReelId         data[15]
 
-static void CreateSlotReelTasks(void)
+static void CreateReelTasks(void)
 {
     u8 i;
     for (i = 0; i < NUM_REELS; i++)
     {
-        u8 taskId = CreateTask(Task_RunSlotReelActions, 2);
+        u8 taskId = CreateTask(Task_Reel, 2);
         gTasks[taskId].tReelId = i;
         sSlotMachine->slotReelTasks[i] = taskId;
-        Task_RunSlotReelActions(taskId);
+        Task_Reel(taskId);
     }
 }
 
 static void SpinSlotReel(u8 reelIndex)
 {
-    gTasks[sSlotMachine->slotReelTasks[reelIndex]].tState = REEL_ACTION_SPIN;
+    gTasks[sSlotMachine->slotReelTasks[reelIndex]].tState = REEL_TASK_SPIN;
     gTasks[sSlotMachine->slotReelTasks[reelIndex]].tMoving = TRUE;
 }
 
 static void StopSlotReel(u8 reelIndex)
 {
-    gTasks[sSlotMachine->slotReelTasks[reelIndex]].tState = REEL_ACTION_STOP;
+    gTasks[sSlotMachine->slotReelTasks[reelIndex]].tState = REEL_TASK_DECIDE_STOP;
 }
 
 static bool8 IsSlotReelMoving(u8 reelIndex)
 {
     return gTasks[sSlotMachine->slotReelTasks[reelIndex]].tMoving;
 }
 
-static void Task_RunSlotReelActions(u8 taskId)
+static void Task_Reel(u8 taskId)
 {
-    while (sSlotReelActions[gTasks[taskId].tState](&gTasks[taskId]))
+    while (sReelTasks[gTasks[taskId].tState](&gTasks[taskId]))
         ;
 }
 
-// task->data[1]  reel turns
-static bool8 SlotReelAction_StayStill(struct Task *task)
+static bool8 ReelTask_StayStill(struct Task *task)
 {
     return FALSE;
 }
 
-static bool8 SlotReelAction_Spin(struct Task *task)
+static bool8 ReelTask_Spin(struct Task *task)
 {
-    AdvanceSlotReel(task->tReelId, sSlotMachine->reelIncrement);
+    AdvanceSlotReel(task->tReelId, sSlotMachine->reelSpeed);
     return FALSE;
 }
 
-// As in previous generations, the slot machine often doesn't stop exactly when you press stop
-static bool8 SlotReelAction_DecideWhereToStop(struct Task *task)
+// In ReelTime, the reel stops immediately. Otherwise, the game may manipulate
+// the results by stopping after at most 4 extra turns. The exact behavior
+// differs depending on whether the machine has a bias.
+//
+// If the machine has a bias, it will try to match the bias tag in each reel.
+//
+// Otherwise, if the machine doesn't have a bias or it could not line up the
+// bias tag in any of the previous reels, it will perform the NoBias stopping
+// routine, which manipulates the outcome so the player loses.
+static bool8 ReelTask_DecideStop(struct Task *task)
 {
-    task->tState++;
-    // initialize data for that reel --> these will be changed if sBiasTags can be lined up
+    task->tState++; // REEL_TASK_STOP_MOVE
     sSlotMachine->winnerRows[task->tReelId] = 0;
     sSlotMachine->reelExtraTurns[task->tReelId] = 0;
 
-    if (sSlotMachine->reelTimeSpinsLeft == 0 && (sSlotMachine->luckyFlags == 0 || !sSlotMachine->isLuckySpin || !sDecideReelTurns_BiasTag[task->tReelId]()))
+    if (sSlotMachine->reelTimeSpinsLeft == 0)
```

Comment:
Why were these conditions separated?

Reply:
I thought it made it a bit clearer that all the logic to manipulate the outcome is skipped in Reel Time. Happy to revert if you'd prefer how it's written currently

Reply:
No that's ok. It seems there's other cases of `sSlotMachine->reelTimeSpinsLeft == 0` as an initial condition by itself, so it's probably more consistent the way you have it now